modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -996,6 +996,45 @@ CV__DNN_EXPERIMENTAL_NS_BEGIN
                              CV_OUT std::vector<int>& indices,
                              const float eta = 1.f, const int top_k = 0);
 
+     class CV_EXPORTS_W Model : public Net
+     {
+     public:
+         Model(const std::string& model, const std::string& config = "");
+         Model(const Net& network);
+
+         Model& setInputSize(const Size& size);
+         Model& setInputSize(int width, int height);
+         Model& setInputMean(const Scalar& mean);
+         Model& setInputScale(float scale);
+         Model& setInputCrop(bool crop);
+         Model& setInputSwapRB(bool swapRB);
+         void predict(InputArray frame, OutputArray outs);
+
+     protected:
+         struct Impl;
+         Ptr<Impl> impl_;
+     };
+
+     class CV_EXPORTS_W ClassificationModel : public Model
+     {
+     public:
+         ClassificationModel(const std::string& model, const std::string& config = "");
+         ClassificationModel(const Net& network);
+
+         std::pair<int, float> classify(InputArray frame);
+     };
+
+     class CV_EXPORTS_W DetectionModel : public Model
+     {
+     public:
+         DetectionModel(const std::string& model, const std::string& config = "");
+         DetectionModel(const Net& network);
+
+         void detect(InputArray frame, CV_OUT std::vector<int>& classIds,
+                     CV_OUT std::vector<float>& confidences, CV_OUT std::vector<Rect2d>& boxes,
+                     float confThreshold = 0.5, float nmsThreshold = 0.4, bool absoluteCoords = true);
```

Comment:
Please add a Python test as well (see https://github.com/opencv/opencv/blob/3.4/modules/dnn/misc/python/test/test_dnn.py).

And additionally please invoke `cv.rectangle` to draw resulting detections on the frame so we can check what rectangles are wrapped correctly.